### ISSUE TITLE
When doing a 'quick' deploy that doesn't change any classes, the deploy fails

### ISSUE THREAD

**Issue by dvf1976** [solenopsis/Solenopsis#211]
There are times when a deployment may change just email templates (or other static artifacts that aren't classes)

When this is done, the deployment fails with the understandable (but avoidable, using the strategy described below) error:
'You must provide at least one test to run'

The workaround is to change a class in some trivial way (adding something to a comment in a class) that Solenopsis then presents at least one test class upon the next deployment.

This is a kludgey workaround :)

Would it be possible to enumerate one 'default' test class that runs in the case when no test classes are enumerated in the artifact that Solenopsis pushes during a 'quick' deploy pre-flight?

This default test class shouldn't be run in addition to all the other test classes Solenopsis discovers.
It should be run only when Solenopsis cannot discover _any_ test classes to run.

I expect there are other strategies that may be used to avoid:
'You must provide at least one test to run'
besides a default test class.

I welcome any strategy that will dodge this error.

**Comment by pcon** [comment]
We should also look at if we do a quick deploy without the test stanza at all in the XML.  I haven't inspected the XML to see if we don't have any runTest stanza.

**Comment by sfloess** [comment]
I don't think using a default test class is the correct answer here.  Instead, to the point @pcon made, we should ignore the ```runTest``` stanza.

**Comment by sfloess** [comment]
@pcon,

Did you have a chance to test this?  I can merge into master when ready sir :)

**Comment by pcon** [comment]
I tested this and the `runtests.xml` looks right (ie doesn't have the `runTest` stanzas).  Unfortunately, I cannot test that it did exactly what I expect because I didn't get a chance to actually run it against production.  My concern is that without the `runTest` stanza that it will deploy it and not stage it for a deployment.  If you don't mind, I'd like to leave this in limbo for a little bit until we have a chance to actually run this against a real production release that does not have any tests.  This way we can verify if it automatically goes to prod or still sits in the deployment queue.

**Comment by sfloess** [comment]
@pcon - sure we can wait on this till you give the word sir :+1:

**Comment by sfloess** [comment]
@pcon - I rebased off master (which now has #205 in it) and did a force-push out remotely...

**Comment by sfloess** [comment]
@pcon - ready for this to merge in?
